nifi-nar-bundles/nifi-standard-bundle/nifi-standard-parameter-providers/src/main/java/org/apache/nifi/parameter/OnePasswordParameterProvider.java

```diff
@@ -0,0 +1,216 @@
+
+                final HttpResponseEntity getVaultItems = webClientServiceProvider.getWebClientService()
+                                                                .get()
+                                                                .uri(URI.create(connectServer + GET_VAULTS + "/" + vaultID + "/items"))
+                                                                .header(AUTHORIZATION_HEADER_NAME, AUTHORIZATION_HEADER_VALUE + accessToken)
+                                                                .header(CONTENT_TYPE, APPLICATION_JSON)
+                                                                .retrieve();
+
+                final JsonNode itemList = OBJECT_MAPPER.readTree(getVaultItems.body());
+                final Iterator<JsonNode> itemIterator = itemList.elements();
+
+                // we iterate though the items
+                while(itemIterator.hasNext()) {
+                    final JsonNode item = itemIterator.next();
+                    final String itemID = item.get("id").asText();
+
+                    final HttpResponseEntity getItemDetails = webClientServiceProvider.getWebClientService()
+                                                                    .get()
+                                                                    .uri(URI.create(connectServer + GET_VAULTS + "/" + vaultID + "/items/" + itemID))
+                                                                    .header(AUTHORIZATION_HEADER_NAME, AUTHORIZATION_HEADER_VALUE + accessToken)
+                                                                    .header(CONTENT_TYPE, APPLICATION_JSON)
+                                                                    .retrieve();
+
+                    final JsonNode itemDetails = OBJECT_MAPPER.readTree(getItemDetails.body());
+                    final String itemName = itemDetails.get("title").asText();
+                    final Iterator<JsonNode> itemFields = itemDetails.get("fields").elements();
+
+                    // we iterate through the fields
+                    while(itemFields.hasNext()) {
+                        final JsonNode field = itemFields.next();
+                        final String fieldId = field.get("id").asText();
+                        final JsonNode fieldValue = field.get("value");
+
+                        if(fieldValue != null) {
+                            final ParameterDescriptor parameterDescriptor = new ParameterDescriptor.Builder().name(itemName + "_" + fieldId).build();
+                            parameters.add(new Parameter(parameterDescriptor, fieldValue.asText(), null, true));
+                        }
+
+                    }
+
+                }
+
+                parameterGroups.add(new ParameterGroup(vaultName, parameters));
+            }
+
+        } catch (final IllegalArgumentException | IOException e) {
+            throw new RuntimeException("Failed to retrieve vault's items", e);
```

Comment:
I wonder if we should have something more specific in place than a RuntimeException, maybe a ParameterFetchException or something, perhaps as a follow on Jira?

Reply:
Yeah that would probably make sense. I took a similar approach as in the other parameter providers we already have.